lib/charms/traefik_k8s/v2/ingress.py

```diff
@@ -651,40 +654,40 @@ def provide_ingress_requirements(
             port: the port of the service (required)
         """
         for relation in self.relations:
-            self._provide_ingress_requirements(scheme, host, port, relation)
+            self._provide_ingress_requirements(scheme, host, ip, port, relation)
 
     def _provide_ingress_requirements(
         self,
         scheme: Optional[str],
         host: Optional[str],
+        ip: Optional[str],
         port: int,
         relation: Relation,
     ):
         if self.unit.is_leader():
             self._publish_app_data(scheme, port, relation)
 
-        self._publish_unit_data(host, relation)
+        self._publish_unit_data(host, ip, relation)
 
     def _publish_unit_data(
         self,
         host: Optional[str],
+        ip: Optional[str],
         relation: Relation,
     ):
         if not host:
             host = socket.getfqdn()
 
         if not ip:
-            network_binding = self.charm.model.get_binding(self.relation)
+            network_binding = self.charm.model.get_binding(relation)
             if network_binding is None:
-                raise RuntimeError(
-                    f"can't find IP address for relation {self.relation}, network binding is None"
-                )
-            binding_ip = network_binding.network.bind_address
-            if binding_ip is None:
-                raise RuntimeError(
-                    f"can't find IP address for relation {self.relation}, bind address is None"
-                )
-            ip = str(binding_ip)
+                ip = None
+            else:
+                binding_ip = network_binding.network.bind_address
+                if binding_ip is None:
+                    ip = None
+                else:
+                    ip = str(binding_ip)
```

Comment:
I think that this can be simplified to:
```python
if network_binding is None and (binding_ip := network_binding.network.bind_address) is None:
   return None
ip = str(binding_ip)
```

Reply:
Gotcha, updated, thanks!